test/integration/node/lifecycle_test.go

```diff
@@ -290,175 +284,94 @@ func TestTaintBasedEvictions(t *testing.T) {
 				true,             // Run taint manager
 			)
 			if err != nil {
-				t.Errorf("Failed to create node controller: %v", err)
-				return
+				t.Fatalf("Failed to create node controller: %v", err)
 			}
 
 			// Waiting for all controllers to sync
 			externalInformers.Start(testCtx.Ctx.Done())
 			externalInformers.WaitForCacheSync(testCtx.Ctx.Done())
-			testutils.SyncInformerFactory(testCtx)
 
-			// Run all controllers
+			// Run the controller
 			go nc.Run(testCtx.Ctx)
-			go testCtx.Scheduler.Run(testCtx.Ctx)
 
 			nodeRes := v1.ResourceList{
 				v1.ResourceCPU:    resource.MustParse("4000m"),
 				v1.ResourceMemory: resource.MustParse("16Gi"),
 				v1.ResourcePods:   resource.MustParse("110"),
 			}
 
+			nodeIndex := 1 // the exact node doesn't matter, pick one
```

Comment:
this should be better placed with `nodeCount`  , since those are related constant for this test , and we can avoid problems in the future if both go out of sync

Reply:
Moved up.